src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -486,7 +486,7 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
                         foreach (var currentPrimary in currentPrimaries)
                         {
                             // kick off the server to invalidate itself
-                            currentPrimary.Invalidate("NoLongerPrimary");
+                            currentPrimary.Invalidate("NoLongerPrimary", args.NewServerDescription.TopologyVersion);
```

Comment:
Due to this line in the [spec commit], (emphasis mine) (https://github.com/mongodb/specifications/commit/7dd9c008be6edaf65368787d5e91c2ac3aeb95b8#diff-62b4da2e38e7d2a9c1a2d92d315895bbR1423-R1425): 
> When the client sees a "not master" or "node is recovering" error and the error's topologyVersion is strictly greater than the current ServerDescription's topologyVersion it MUST replace the server's description with a ServerDescription of type Unknown. Clients MUST store useful information in the new ServerDescription's error field, including the error message from the server. **Clients MUST store the error's topologyVersion field in the new ServerDescription if present**.